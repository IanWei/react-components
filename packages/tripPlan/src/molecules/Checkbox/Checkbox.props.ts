export interface CheckboxProps {
    label: string,
    handleCheckboxChange(label: string): void
}
