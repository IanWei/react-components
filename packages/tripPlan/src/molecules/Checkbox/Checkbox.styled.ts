import styled from 'styled-components';
import { fontSize } from '../../foundation/typography';
import { device } from '../../foundation/breakpoints';
import { spacing } from '../../foundation/spacing';

export const StyledCheckboxWrapper = styled.div`
    display: flex;
    align-items: center;
    user-select: none;
    
    &:not(:last-of-type) {
        margin-bottom: ${spacing.xxxs};    
    }
`
export const StyledLabel = styled.label`
    position: relative;
    display: flex;
    align-items: center;
    font-size: ${fontSize.xs};
    line-height: ${fontSize.xs};
    padding-left: 1.15rem;
    cursor: pointer;
    
    @media only screen and ${device.tablet} {
        font-size: ${fontSize.sm};
    }
    
    @media only screen and ${device.desktop} {
        font-size: ${fontSize.base};
    }
`
export const StyledInput = styled.input`
    position: absolute;
    top: 50%;
    left: 0;
    transform: translateY(-50%);
`
