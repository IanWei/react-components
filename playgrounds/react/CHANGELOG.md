# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.2.0](https://github.com/IanWei/react-components/compare/v0.1.1...v0.2.0) (2021-07-04)


### Features

* calendar component ([24182ce](https://github.com/IanWei/react-components/commit/24182ce40d531de8a861fa57808c2b16e24e54f9)), closes [#031](https://github.com/IanWei/react-components/issues/031)
* calendar continue ([38cd923](https://github.com/IanWei/react-components/commit/38cd9238ce43aef2df0acaf1fb614f53809359bf)), closes [#033](https://github.com/IanWei/react-components/issues/033)
* keyboard acceisible ([929915f](https://github.com/IanWei/react-components/commit/929915fd32803440d14a673b550c575a0bc37828)), closes [#028](https://github.com/IanWei/react-components/issues/028)
* navigate component ([fd3c74a](https://github.com/IanWei/react-components/commit/fd3c74aa7ed9e52c12b1b904a7658caa8ec1f119)), closes [#032](https://github.com/IanWei/react-components/issues/032)
* update Select component ([356c718](https://github.com/IanWei/react-components/commit/356c718b96c5675da9d345a9b8aee2c369a3af9a)), closes [#027](https://github.com/IanWei/react-components/issues/027)
