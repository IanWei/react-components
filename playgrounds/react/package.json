{
  "name": "@playground/react",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "scripts": {
    "dev": "parcel src/index.html -p 3000"
  },
  "devDependencies": {
    "@types/react": "^17.0.13",
    "@types/react-dom": "^17.0.8",
    "parcel-bundler": "^1.12.5",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "typescript": "^4.3.5"
  },
  "private": true
}
