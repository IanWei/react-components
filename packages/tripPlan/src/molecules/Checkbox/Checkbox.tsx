import React, { FC, useState } from 'react';
import { StyledCheckboxWrapper, StyledInput, StyledLabel } from './Checkbox.styled';
import { CheckboxProps } from './Checkbox.props';

export const Checkbox: FC<CheckboxProps> = ({ label, handleCheckboxChange }) => {
    const [isChecked, setCheck] = useState<boolean>(false);
    const toggleCheckboxChange = () => {
        setCheck(!isChecked);
        handleCheckboxChange(label)
    }
    return (
        <StyledCheckboxWrapper>
            <StyledLabel>
                <StyledInput type="checkbox" checked={isChecked} onChange={toggleCheckboxChange}/>
                {label}
            </StyledLabel>
        </StyledCheckboxWrapper>
    )
}
