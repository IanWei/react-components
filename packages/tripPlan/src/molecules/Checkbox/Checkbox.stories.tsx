import React from 'react';
import { Checkbox } from './Checkbox';
import { CheckboxProps } from './Checkbox.props';
import { Story } from '@storybook/react';
import { text } from '@storybook/addon-knobs';

export default {
    title: 'Molecules/Checkbox',
    component: Checkbox
};

const Template: Story<CheckboxProps> = (args) => (
    <Checkbox {...args} label={text('Label', args.label)} />
);

export const Primary: Story<CheckboxProps> = Template.bind({});

Primary.args = {
    label: 'Testing',
    handleCheckboxChange(label: string) {
        console.log(label);
    }
};
