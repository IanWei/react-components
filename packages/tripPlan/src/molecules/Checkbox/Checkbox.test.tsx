import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Checkbox } from './Checkbox';

describe('Checkbox', () => {
    let checkbox: HTMLDivElement;
    let snapshot: () => DocumentFragment;
    const handleCheckboxChange = jest.fn();

    beforeEach(() => {
        const { queryByTitle, asFragment } = render(
            <Checkbox
                label="test"
                handleCheckboxChange={handleCheckboxChange}
            />
        );
        checkbox = queryByTitle('checkbox') as HTMLDivElement;
        snapshot = asFragment;
    });

    it('should render checkbox', () => {
        expect(checkbox).toBeTruthy();
    });

    it('should render text passed by label prop', () => {
        expect(screen.getByRole('label')).toHaveTextContent('test');
    });

    it('should call the handleCheckboxChange prop with the checkbox while it is toggled', () => {
        fireEvent.click(checkbox);
        expect(handleCheckboxChange).toHaveBeenCalled();
    });

    it('should test the snapshot of the base state', () => {
        expect(snapshot()).toMatchSnapshot();
    });

    it('should test the snapshot of the toggle state', () => {
        fireEvent.click(checkbox);
        expect(snapshot()).toMatchSnapshot();
    });
});
