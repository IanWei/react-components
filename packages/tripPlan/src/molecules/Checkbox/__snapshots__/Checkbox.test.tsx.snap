// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should test the snapshot of the base state 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM cLofpq"
  >
    <label
      class="sc-gtsrHT eqeuFw"
      title="option"
    >
      <input
        aria-checked="false"
        aria-label="test"
        class="sc-dlnjwi bUvIpl"
        role="checkbox"
        type="checkbox"
      />
      test
    </label>
  </div>
</DocumentFragment>
`;

exports[`Checkbox should test the snapshot of the toggle state 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM cLofpq"
  >
    <label
      class="sc-gtsrHT eqeuFw"
      title="option"
    >
      <input
        aria-checked="true"
        aria-label="test"
        class="sc-dlnjwi bUvIpl"
        role="checkbox"
        type="checkbox"
      />
      test
    </label>
  </div>
</DocumentFragment>
`;
